packages/wallet/src/domain/two-player-consensus-game/index.ts

```diff
@@ -0,0 +1,199 @@
+export function encodeAppAttributes(appAttrs: AppAttributes): Bytes {
+  return abi.encodeParameter(SolidityConsensusCommitmentType, [
+    appAttrs.consensusCounter,
+    appAttrs.proposedAllocation,
+    appAttrs.proposedDestination,
+  ]);
+}
+
+export function decodeAppAttributes(appAttrs: Bytes): AppAttributes {
+  return appAttributes(abi.decodeParameter(SolidityConsensusCommitmentType, appAttrs));
+}
+
+export function asCoreCommitment(commitment: ConsensusGameBaseCommitment): Commitment {
+  const {
+    channel,
+    commitmentType,
+    turnNum,
+    allocation,
+    destination,
+    commitmentCount,
+    consensusCounter,
+    proposedAllocation,
+    proposedDestination,
+  } = commitment;
+
+  return {
+    channel,
+    commitmentType,
+    turnNum,
+    allocation,
+    destination,
+    commitmentCount,
+    appAttributes: encodeAppAttributes({
+      consensusCounter,
+      proposedAllocation,
+      proposedDestination,
+    }),
+  };
+}
+
+export function fromCoreCommitment(commitment: Commitment): ConsensusGameBaseCommitment {
+  const { channel, turnNum, commitmentType, allocation, destination, commitmentCount } = commitment;
+  const { consensusCounter, proposedAllocation, proposedDestination } = decodeAppAttributes(
+    commitment.appAttributes,
+  );
+  return {
+    channel,
+    turnNum,
+    allocation,
+    destination,
+    commitmentCount,
+    commitmentType,
+    consensusCounter,
+    proposedAllocation,
+    proposedDestination,
+  };
+}
+
+// TODO
+// function initialize(channelDetails: ??, initialOutcome: Outcome): Proposal
+
+export function accept(commitment: ProposalCommitment): ConsensusCommitment {
+  return consensusCommitment({
+    ...commitment,
+    turnNum: commitment.turnNum + 1,
+    allocation: commitment.proposedAllocation,
+    destination: commitment.proposedDestination,
+    // TODO should proposedAllocation be conserved, or reset?
+  });
+}
+
+export function reject(commitment: ProposalCommitment): ConsensusCommitment {
+  return consensusCommitment({
+    ...commitment,
+    turnNum: commitment.turnNum + 1,
+    allocation: commitment.allocation,
+    destination: commitment.destination,
+    // TODO should proposedAllocation be conserved, or reset?
+  });
+}
+
+export function propose(
+  commitment: ConsensusCommitment,
+  proposedAllocation: Uint256[],
+  proposedDestination: Address[],
+): ProposalCommitment {
+  return proposalCommitment({
+    ...commitment,
+    turnNum: commitment.turnNum + 1,
+    proposedAllocation,
+    proposedDestination,
+    // TODO should proposedAllocation be conserved, or reset?
+  });
+}
+
+export function isConsensus(
+  commitment: ConsensusGameBaseCommitment,
+): commitment is ConsensusCommitment {
+  return commitment.consensusCounter === 0;
+}
+export function isProposal(
+  commitment: ConsensusGameBaseCommitment,
+): commitment is ProposalCommitment {
+  return commitment.consensusCounter === 1;
+}
+
+export function nextLedgerUpdateCommitment(
```

Comment:
I'm not sure we want this method. I'd prefer to always ensure we use the methods above in the code. Could you remove this and replace the places where it's called?